lib/puppet/pops/parser/lexer2.rb

```diff
@@ -491,9 +493,9 @@ def initialize()
       '' => lambda { nil } # when the peek(1) returns empty
     }
 
-    [ ' ', "\t", "\r" ].each { |c| @selector[c] = lambda { @scanner.skip(PATTERN_WS); nil } }
+    [' ', "\t", "\r"].each { |c| @selector[c] = lambda { @scanner.skip(PATTERN_WS); nil } }
 
-    [ '0', '1', '2', '3', '4', '5', '6', '7', '8', '9'].each do |c|
+    ('0'..'9').each do |c|
```

Comment:
More thinking out loud, are there perf implications to this? We generally have to be careful about changes to the lexer, even simple things like referencing an instance variable vs a local variable. It might be worth creating a sample test and running it in benchmark-ips with JRuby 9.4 and see if there's any appreciable difference?

Reply:
Running the benchmark here: https://github.com/puppetlabs/puppet/blob/97f785494999c358844526871e55fb8c39c10059/spec/unit/pops/benchmark_spec.rb#L83-L87

The performance seems comparable between this branch and main, both with MRI and JRuby:

| Environment                                     | Results                                               |
| ----------------------------------------------- | ----------------------------------------------------- |
| MRI 3.2.2, main                                 | Lexer2:   0.245682   0.016103   0.261785 (  0.264757) |
| MRI 3.2.2, PUP-11768/main/layout-and-cops-6     | Lexer2:   0.242654   0.014749   0.257403 (  0.261339) |
| JRuby 9.4.3.0,main                              | Lexer2:   2.540000   0.040000   2.580000 (  0.799291) |
| JRuby 9.4.3.0, PUP-11768/main/layout-and-cops-6 | Lexer2:   2.380000   0.030000   2.410000 (  0.766099) |